server/src-lib/Hasura/Metadata/Class.hs

```diff
@@ -15,6 +16,7 @@ import           Hasura.RQL.Types
 
 import qualified Hasura.Tracing                         as Tracing
 
+-- | Metadata storage transformer that enables raising @'QErr' exceptions.
```

Comment:
It's clear from the definition 2 lines below this comment that it enables raising `QErr` exceptions. The question to answer is _why_ we want to raise `QErr`ors, and if this monad transformer represents anything else in addition to that.